src/macros.rs

```diff
@@ -1,47 +1,250 @@
-/// Construct a BSON value
+// BSON macro based on the serde_json json! implementation.
+
+/// Construct a bson::BSON value from a literal.
+///
+/// ```rust
+/// # #[macro_use]
+/// # extern crate bson;
+/// #
+/// # fn main() {
+/// let value = bson!({
+///     "code": 200,
+///     "success": true,
+///     "payload": {
+///       "some": [
+///           "pay",
+///           "loads",
+///       ]
+///     }
+/// });
+/// # }
+/// ```
 #[macro_export]
 macro_rules! bson {
-    ([]) => {{ $crate::Bson::Array(Vec::new()) }};
+    //////////////////////////////////////////////////////////////////////////
```

Comment:
I assume these comments are also lifted from serde_json?

Reply:
yeah; not sure where the line falls in terms of fair usage.